### PR TITLE
Ports several more smites, minorly refactors two existing smites

### PR DESCRIPTION
**Pull request by RipGrayson**
Ports two smites that didn't make it into the first PR, adds one original smite. 
Changes several smites to have less shitcode.

## About The Pull Request

During the first PR several smites had to be dropped to due falling outside the specification due to complexity. This PR restores them. 
Also implements misc. changes to existing smites, improving usability and fixing an error where the first option of bloodloss was essentially worthless.

Edit: credit for the original two smites being ported goes to
Ryll-Ryll for the [shoe knotting smite](https://github.com/tgstation/tgstation/pull/52157)
and Anturk for the [breadify smite](https://github.com/tgstation/tgstation/pull/53180)
## Why It's Good For The Game
More smites to ~~entertain the admins~~ punish rule breaking players in creative ways.

## Changelog
:cl:
expansion: Added two more smites to the smite system
fix: The first option in bloodloss now actually does something
refactor: Changed some smite code to be slightly better
/:cl:

### CODE REVIEW

File: code/modules/admin/smites/bread.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#define BREADIFY_TIME (5 SECONDS)
+#define BREAD_PARALYZE_TIME (5 MINUTES)
+
+/// Turns the target into bread
+/datum/smite/bread
+	name = "Bread"
+
+/datum/smite/bread/effect(client/user, mob/living/target)
+	. = ..()
+	var/confirm = tgui_alert(usr, "Irreversibly transform [target] into a delectable loaf of bread?", "Continue?", list("Yes", "No"))
```
Comment (BraveMole) [inline code/modules/admin/smites/bread.dm:10]:
```suggestion
	if(tgui_alert(usr, "Irreversibly transform [target] into a delectable loaf of bread?", "Continue?", list("Yes", "No")) != "Yes")
		return
```
else closing the window will be counted as a yes

File: code/modules/admin/smites/knot_shoes.dm [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	var/obj/item/clothing/shoes/sick_kicks = target.shoes //our target's shoes
+	if (!sick_kicks?.can_be_tied)
+		to_chat(user, span_warning("[target] does not have knottable shoes!"), confidential = TRUE)
+		return
+	if (sick_kicks.tied == SHOES_TIED) //if tied knot the shoes of our victim
+		sick_kicks.adjust_laces(SHOES_KNOTTED) 
+	else //If we're already knotted, revert them back to being tied. Marines aren't smart enough to unknot their own shoes without divine intervention.
+		var/confirm = tgui_alert(usr, "Unknot the shoes of [target]?", "Continue?", list("Yes", "No"))
+		if(confirm == "No")
+			return
```
Comment (BraveMole) [inline code/modules/admin/smites/knot_shoes.dm:19]: ditto

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/admin/smites/headset_explosion.dm [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/// explode target headset dealing head brain and eye damage
+/datum/smite/headset_explosion
+	name = "Headset Explosion"
+
+/datum/smite/headset_explosion/effect(client/user, mob/living/carbon/human/target)
```
Comment (BraveMole) [inline code/modules/admin/smites/headset_explosion.dm:5]: is this a tg port?

Comment (RipGrayson) [reply code/modules/admin/smites/headset_explosion.dm:5]: This one is an original, inspired by the time everybody received an exploding headset for a round. As such all shitcode in this smite is 100% on me.

File: code/modules/admin/smites/headset_explosion.dm [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	if(!ishuman(target))
+		to_chat(user, span_warning("Target does not have ears. Aborting."), confidential = TRUE)
+		return
+	if(target.wear_ear == null)
+		to_chat(user, span_warning("Target does not have a headset. Aborting."), confidential = TRUE)
+		return
+
+	var/datum/limb/L = target.get_limb("head")
+	var/confirm = tgui_alert(usr, "Blow [target]'s head off with the explosion?", "Explosion Strength", list("Yes", "No"))
+	if(confirm == "Yes")
```
Comment (BraveMole) [inline code/modules/admin/smites/headset_explosion.dm:17]: no need to create a new var here

**Comment by RipGrayson** [comment]
> can you credit the prs youre porting?

This a continuation of my efforts to port TG's smite system, unless you want me to credit the last one to rework that system it's not something that can be credited as a single PR. 

If you mean the two TG smites being ported, knot shoes originally came from here; https://github.com/tgstation/tgstation/pull/52157, and breadify came from here https://github.com/tgstation/tgstation/pull/53180. Both have since been altered by both TG reworks and my own changes to make them TGMC compatible.

I could credit the original authors if you want.

**Comment by RipGrayson** [comment]
> It's polite to credit code authors yes, and that's a thing you should always do in the open source environnment. The way you can usually do it is by linking the ported PRs, which also allows reviewerers to check that you are indeed porting something that works

Now credited in the opening of the PR

File: code/game/objects/items/reagent_containers/food/snacks.dm [hunk 5]
```diff
@@ -2074,6 +2074,19 @@
...
+	icon_state = "breadtg"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 60)
+	filling_color = "#FFF896"
+	bitesize = 2
+	tastes = list("guilt" = 1, "salt" = 1)
+
+/obj/item/reagent_containers/food/snacks/marinebread/Destroy() //delete the marine trapped inside, tasty!
+	for(var/i in contents)
+		qdel(i)
+	return ..()
```
Comment (TiviPlus) [inline code/game/objects/items/reagent_containers/food/snacks.dm:2089]: this is already done on /atom/movable/Destroy

File: code/modules/admin/smites/boneless.dm [hunk 6]
```diff
@@ -8,9 +8,11 @@
...
-		
+	
+	var/timer = 2 SECONDS
 	to_chat(target,span_userdanger("Your bones break in a spray of blood, sending bone fragments everywhere!"))
 	for(var/datum/limb/limb_to_break AS in target.limbs)
 		if(limb_to_break.limb_status & (LIMB_BROKEN | LIMB_DESTROYED | LIMB_AMPUTATED))
 			continue
-		limb_to_break.fracture()
+		addtimer(CALLBACK(limb_to_break, /datum/limb.proc/fracture), timer - 0.4 SECONDS)
+		timer += 2 SECONDS
```
Comment (TiviPlus) [inline code/modules/admin/smites/boneless.dm:18]: Mistype with the timer being after the timer?

File: code/modules/admin/smites/headset_explosion.dm (hunk 3 above)
Comment (TiviPlus) [reply code/modules/admin/smites/headset_explosion.dm:5]: you shouldnt put the new smite with the smite ports split it out please

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/items/reagent_containers/food/snacks.dm (hunk 5 above)
Comment (RipGrayson) [reply code/game/objects/items/reagent_containers/food/snacks.dm:2089]: By default reagent_containers/food/snacks/Destroy() will place any items inside on the floor. The breadify smite qdels the human target on regular TG so I overrode /food/snacks/Destroy() to get that behavior.
If there's a better way of doing it I'm all ears.

Comment (RipGrayson) [reply code/game/objects/items/reagent_containers/food/snacks.dm:2089]: I guess I could delete the Destroy() that dropped contents to turf, but I was trying to preserve the original behavior.

**Review by TiviPlus — APPROVED** [review]

**Comment by TiviPlus** [comment]
@Kuro020

**Review by Kuro020 — APPROVED** [review]

**Pull request merged**